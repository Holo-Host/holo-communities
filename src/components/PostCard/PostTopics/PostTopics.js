import React from 'react'
import { Link } from 'react-router-dom'
import { tagUrl } from 'util/navigation'
import './PostTopics.scss'
import cx from 'classnames'

export default function PostTopics ({ topics, slug, spacer }) {
  const Spacer = spacer

  return <div styleName={cx('topicsLine', { 'newLineForTopics': !spacer })}>
    {spacer && <Spacer />}
    {topics.slice(0, 3).map(t =>
      <Link styleName='topic' to={tagUrl(t.name, slug)} key={t.name}>#{t.name}</Link>)}
  </div>
}
