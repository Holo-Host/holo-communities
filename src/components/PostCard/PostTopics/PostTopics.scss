.topicsLine {
  composes: text-button from 'css/typography.scss';
  font-size: 12px;
  display: inline-block;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  max-width: 220px;
}

.topic {
  composes: text-button from 'css/typography.scss';
  font-size: 12px;
  margin-right: 10px;
}

.newLineForTopics {
  overflow: initial;
  text-overflow: initial;
  margin-top: -7px;
  padding-left: 1rem;
  margin-bottom: 10px;
  width: 450px;
}
