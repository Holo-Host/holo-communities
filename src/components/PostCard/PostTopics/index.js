import component from './PostTopics'

export default component
