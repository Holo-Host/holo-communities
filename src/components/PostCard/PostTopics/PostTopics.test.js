import PostTopics from './PostTopics'
import { shallow } from 'enzyme'
import React from 'react'

describe('TopicsLine', () => {
  it('matches last snapshot', () => {
    const props = {
      topics: [{ name: 'one' }, { name: 'two' }],
      slug: 'hay',
      spacer: () => <span>.</span>
    }

    const wrapper = shallow(<PostTopics {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
