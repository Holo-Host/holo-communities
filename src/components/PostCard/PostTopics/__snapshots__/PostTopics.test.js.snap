// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopicsLine matches last snapshot 1`] = `
<div
  data-stylename="topicsLine"
>
  <spacer />
  <Link
    data-stylename="topic"
    key="one"
    replace={false}
    to="/c/hay/one"
  >
    #
    one
  </Link>
  <Link
    data-stylename="topic"
    key="two"
    replace={false}
    to="/c/hay/two"
  >
    #
    two
  </Link>
</div>
`;
